### ISSUE TITLE
Cursor appears several characters off

### ISSUE THREAD

**Issue by maryfries** [jmoenig/Snap#2395]
![snap-cursor-issues](https://user-images.githubusercontent.com/6549787/57794081-d3e9ae80-7710-11e9-8a0c-a546eb6a9535.gif)

**Comment by bromagosa** [comment]
Did you by any chance type any funny UTF-8 characters, like emoji or other non-ascii chars, before the text you're showing?

**Comment by maryfries** [comment]
No, but I did have a very long line of text and this had happened previously:

![image](https://user-images.githubusercontent.com/6549787/57799886-149bf480-771e-11e9-8e5d-94d1877622b0.png)

I first typed hue/saturation/brightness/transparency and then I decided to switch to caps and started typing HUE/SATURATION/BRIGHTNESS/TRANSPARENCY in front of it without a space. Then the comment fell apart as pictured here, and later the cursor started misbehaving.

**Comment by jguille2** [comment]
Yes, I can reproduce this (in FF and Chrome too)

The problem is when you type a "word" (string without withespaces inside) longer than the comment bubble width. Text is dynamically moved to the left (out of the bubble) and the issue begins.

In contrast, if you make a comment bubble shorter (changing its size) and then, some "words" become longer than the width, the problem does not appear, and the "word" is cut.

Joan
